packages/framework/src/Model/Article/ArticleDetailFriendlyUrlDataProvider.php

```diff
@@ -54,9 +54,20 @@ public function getFriendlyUrlData(DomainConfig $domainConfig): array
         $friendlyUrlsData = [];
 
         foreach ($scalarData as $data) {
-            $friendlyUrlData = $this->friendlyUrlDataFactory->create();
-            $friendlyUrlData->name = $data['id'];
-            $friendlyUrlData->id = $data['name'];
+            if (method_exists($this->friendlyUrlDataFactory, 'createFromIdAndName')) {
+                $friendlyUrlData = $this->friendlyUrlDataFactory->createFromIdAndName($data['id'], $data['name']);
+            } else {
+                @trigger_error(
+                    sprintf(
+                        'This approach is deprecated since Shopsys Framework 7.3, implement and use "%s" instead',
+                        FriendlyUrlDataFactoryInterface::class . '::createFromIdAndName()'
+                    ),
+                    E_USER_DEPRECATED
+                );
+                $friendlyUrlData = $this->friendlyUrlDataFactory->create();
+                $friendlyUrlData->id = $data['id'];
+                $friendlyUrlData->name = $data['name'];
```

Comment:
Why don't you use `createFromData` method? You can omit the deprecations here and have it only in `FriendlyUrlDataFactory`

Reply:
I was not able to use it as it is not part of the interface...But your point makes sense, rewritten by adding the @method annotation along with @deprecated sign into the interface :grin:   :+1: